src/util.ts

```diff
@@ -62,25 +63,27 @@ export function fractionBN(
   return targetBN.mul(numBN).div(denomBN);
 }
 
-const BN_1GWEI_IN_WEI = new BN(1000000000, 10);
-
 /**
  * Used to convert a base-10 number from GWEI to WEI. Can handle numbers with decimal parts
  *
  * @param n - The base 10 number to convert to WEI
  * @returns - The number in WEI, as a BN
  */
 export function gweiDecToWEIBN(n: number | string) {
-  const wholePart = Math.floor(Number(n));
-  const decimalPartMatch = Number(n)
-    .toFixed(9)
-    .match(/\.(\d+)/u);
-  const decimalPart = decimalPartMatch ? decimalPartMatch[1] : '0';
-
-  const wholePartAsWEI = new BN(wholePart, 10).mul(BN_1GWEI_IN_WEI);
-  const decimalPartAsWEI = new BN(decimalPart, 10);
+  if (Number.isNaN(n)) {
+    return new BN(0);
+  }
+  return toWei(n.toString(), 'gwei');
```

Comment:
It looks like this now throws an error if you input a number with too many decimal places. Beforehand it would ignore the extra decimals instead.

This seems safer. I just wanted to note the change in behaviour.